rpc/src/server/service.rs

```diff
@@ -406,6 +446,197 @@ mod fns {
             .map(|values| values.into_iter().map(|v| v.action).collect())
             .map_err(|e| e.into())
     }
+    pub(crate) fn get_block_level(block_id: &str, persistent_storage: &PersistentStorage, list: &ContextList) -> Result<Option<usize>, failure::Error> {
+        if block_id == "head" {
+            let reader = list.read().expect("mutex poisoning");
+            Ok(Some(reader.len() - 1))
+        } else {
+            let block_hash = HashType::BlockHash.string_to_bytes(block_id)?;
+            let block_meta_storage: BlockMetaStorage = BlockMetaStorage::new(persistent_storage);
+            if let Some(block_meta) = block_meta_storage.get(&block_hash)? {
+                Ok(Some(block_meta.level() as usize))
+            } else {
+                Ok(None)
+            }
+        }
+    }
+
+    pub(crate) fn get_rolls(block_id: &str, _cycle: &String, persistent_storage: &PersistentStorage, list: ContextList) -> Result<Option<HashMap<u32, String>>, failure::Error> {
+        // get the level of from the block
+        let level = match get_block_level(block_id, persistent_storage, &list).unwrap() {
+            Some(v) => v,
+            None => return Ok(None)  // Cannot get level
+        };
+
+        // TODO: implement getting the random seed out of context storage
+        //       based on the cycle 
+
+        // get the whole context
+        let context = {
+            let reader = list.read().unwrap();
+            if let Ok(Some(ctx)) = reader.get(level) {
+                ctx
+            } else {
+                panic!("Context not found")
+            }
+        };
+
+        let roll_data = context.clone();
+        // get all the relevant data out of the context database
+        let data: HashMap<String, Bucket<Vec<u8>>> = context.into_iter()
+            .filter(|(k, _)| k.contains("data/rolls/owner/current") || k.contains("/successor") || k.contains("/random_seed"))
```

Comment:
As an improvement, also add a check, that value is `Bucket::Exists` allowing safe `unwrap`s down the code.

Reply:
When accessing the created hash maps from the context, I always perform a check whether the value is a Bucket::Exists variant. (E.g [here](https://github.com/adonagy/tezedge/blob/baking_rights/rpc/src/server/service.rs#L515)). Should I add the check nevertheless?